src/xdp-utils.c

```diff
@@ -2349,3 +2349,67 @@ xdp_validate_serialized_icon (GVariant  *v,
 
   return TRUE;
 }
+
+#if G_ENCODE_VERSION (GLIB_MAJOR_VERSION, GLIB_MINOR_VERSION) >= G_ENCODE_VERSION (2, 66)
+gchar *
+xdp_transform_remote_uri_into_local (const gchar *uri)
+{
+  gchar *pathtext = NULL;
+  gchar *uristring = NULL;
+  gchar *fuse_mountpoint = NULL;
+  GUri *file_uri = NULL;
+  GString *gvfs_folder = NULL;
+
+  file_uri = g_uri_parse (uri, G_URI_FLAGS_NONE, NULL);
+  if (file_uri == NULL)
+    {
+      return g_strdup(uri);
+    }
+  if ((0 != g_strcmp0 (g_uri_get_scheme (file_uri), "file")) &&
+      (g_uri_get_host (file_uri) != NULL))
+    {
+      if (g_get_user_runtime_dir() == g_get_user_cache_dir ())
```

Comment:
Use g_strcmp0 here? Comparing two pointers doesn't seem to be what you want here.

Reply:
OK, not so rookie... I just copy-pasted the code from Gvfs...

![imagen](https://user-images.githubusercontent.com/105285003/181219244-069ffb4e-60ef-44e1-adba-f1d7b968bf6a.png)

Should I send a patch?

Reply:
Yes, you should probably open a MR for that in gvfs. This code may work because the pointers happen to be the same, but it seems like even if it works it could break at any point if glib changes the way it stores these values.

Reply:
Sent https://gitlab.gnome.org/GNOME/gvfs/-/merge_requests/151